src/shogun/base/unique.h

```diff
@@ -0,0 +1,44 @@
+#ifndef __SG_UNIQUE_H__
+#define __SG_UNIQUE_H__
+
+#ifdef HAVE_CXX11
+
+#include <memory>
+
+namespace shogun
```

Comment:
We should also avoid SWIG from going through this file, right? I am not sure if that is already happening since there is no mention to this in any .i file or we should also use the guard.